apps/api/src/domain/models/categories/types/ResourceTypeOrNoteType.ts

```diff
@@ -0,0 +1,11 @@
+import { isResourceType, ResourceType } from '../../../types/resourceTypes';
+
+export const noteType = 'note';
+
+export type NoteType = typeof noteType;
+
+// TODO improve naming. What is the significance of this in the domain?
+export type ResourceTypeOrNoteType = ResourceType | NoteType;
```

Comment:
Find a new name like "TaggableEntities"

Reply:
I'll make a story for this as I want to experiment with 'extending' the existing `resourceTypes` enum.